src/scaffoldmaker/meshtypes/meshtype_3d_bladder1.py

```diff
@@ -880,13 +573,676 @@ def defineFaceAnnotations(cls, region, options, annotationGroups):
+        lumenOfBody_ventral.getMeshGroup(mesh2d).addElementsConditional(is_body_lumen_ventral)
+
+        is_neck_serosa_dorsal = fm.createFieldAnd(is_neck_serosa, is_dorsal_bladder)
+        serosaOfNeck_dorsal = \
+            findOrCreateAnnotationGroupForTerm(annotationGroups, region,
+                                               get_bladder_term("dorsal part of serosa of neck of urinary bladder"))
+        serosaOfNeck_dorsal.getMeshGroup(mesh2d).addElementsConditional(is_neck_serosa_dorsal)
+
+        is_neck_serosa_ventral = fm.createFieldAnd(is_neck_serosa, is_ventral_bladder)
+        serosaOfNeck_ventral = \
+            findOrCreateAnnotationGroupForTerm(annotationGroups, region,
+                                               get_bladder_term("ventral part of serosa of neck of urinary bladder"))
+        serosaOfNeck_ventral.getMeshGroup(mesh2d).addElementsConditional(is_neck_serosa_ventral)
+
+        is_neck_lumen_dorsal = fm.createFieldAnd(is_neck_lumen, is_dorsal_bladder)
+        lumenOfNeck_dorsal = \
+            findOrCreateAnnotationGroupForTerm(annotationGroups, region,
+                                               get_bladder_term("dorsal part of lumen of neck of urinary bladder"))
+        lumenOfNeck_dorsal.getMeshGroup(mesh2d).addElementsConditional(is_neck_lumen_dorsal)
+
+        is_neck_lumen_ventral = fm.createFieldAnd(is_neck_lumen, is_ventral_bladder)
+        lumenOfNeck_ventral = \
+            findOrCreateAnnotationGroupForTerm(annotationGroups, region,
+                                               get_bladder_term("ventral part of lumen of neck of urinary bladder"))
+        lumenOfNeck_ventral.getMeshGroup(mesh2d).addElementsConditional(is_neck_lumen_ventral)
+
+def findDerivativeBetweenPoints(v1, v2):
+    """
+    Find vector difference between two points and rescale vector difference using cubic hermite arclength
+    between the points to derive the derivative between the points.
+    :param v1: start vector
+    :param v2: end vector
+    :return: derivative of between v1 and v2
+    """
+    d = [v2[c] - v1[c] for c in range(3)]
+    arcLengthAround = interp.computeCubicHermiteArcLength(v1, d, v2, d, True)
+    d = [c * arcLengthAround for c in vector.normalise(d)]
+
+    return d
+
+def findNodesAlongBladderDome(sx_dome_group, elementsCountAround, elementsCountAlongDome):
+
+    # Find apex nodes d2
+    d2Apex = []
+    d2 = sx_dome_group[2][0]
+    for n1 in range(elementsCountAround):
+        rotAngle = n1 * 2.0 * math.pi / elementsCountAround
+        rotAxis = vector.normalise(vector.crossproduct3(vector.normalise(sx_dome_group[2][0]), vector.normalise(sx_dome_group[4][0])))
+        rotFrame = matrix.getRotationMatrixFromAxisAngle(rotAxis, rotAngle)
+        d2Rot = [rotFrame[j][0] * d2[0] + rotFrame[j][1] * d2[1] + rotFrame[j][2] * d2[2] for j in range(3)]
+        d2Apex.append(d2Rot)
```

Comment:
Having constant angle around the apex is not strictly correct for ellipsoid shape, but let's see how it looks with the other suggested fixes.

Reply:
It seems working properly.